Allow customer to specify either from or to number

zhizhongk opened the pull request:
*Issue #, if available:*
This is to allow customer to enter either fromNumber or toNumber to establish the connection for transcription transmission

*Description of changes:*
1. Add the connection table to store connection-number pair. This table is going to help release resource when client disconnects the socket. 
2. API Gateway integration function changed to put item into connection table. Also the function starts to handler the disconnection(resource releasing)
3. Publisher will check the fromNumber and toNumber respectively in the mapping table. If both cannot be found then connection cannot be established.

Minor:
1. Remove provisioned concurrency. During the testing it is unstable and sometimes times out for unknown reason.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

rajpathi commented on src/main/java/com/amazonaws/kvstranscribestreaming/lambda/TranscriptionWebSocketIntegrationLambda.java:
```diff
@@ -44,33 +57,91 @@ public APIGatewayV2WebSocketResponse handleRequest(APIGatewayV2WebSocketEvent re
+                return responseEvent;
+            }
+
+            DynamoDB dynamoDB = new DynamoDB(AmazonDynamoDBClientBuilder.standard().withRegion(AWS_REGION.getName()).build());
```
Don't initialize it for every request.

zhizhongk commented on src/main/java/com/amazonaws/kvstranscribestreaming/publisher/WebSocketTranscriptionPublisher.java:
```diff
@@ -145,12 +158,13 @@ private String buildTranscription(Result result) {
                 nf.format(result.startTime()),
                 nf.format(result.endTime()),
-                this.detail.getIsCaller() == Boolean.TRUE ? "spk_0" : "spk_1",
+                this.detail.getIsCaller() == Boolean.TRUE ? callerLabel : calleeLabel,
```
The actual value for these two variables are here https://github.com/aws-samples/amazon-chime-voiceconnector-transcription/pull/24/files#diff-a6e5b08a65259775a5973856deba2d5eR161

zhizhongk commented:
This commit has been fully tested and proved working well in my personal account. I have addressed all the comments.

Because of limited bandwidth and urgent need to release a version, I will override and merge the pull request
